src/OpenColorIO/GpuShaderDesc.cpp

```diff
@@ -74,12 +74,27 @@ class GpuShaderCreator::Impl
             m_functionHeader = rhs.m_functionHeader;
             m_functionBody   = rhs.m_functionBody;
             m_functionFooter = rhs.m_functionFooter;
+            
+            updateClassWrappingInterface();
+            *m_classWrappingInterface = *rhs.m_classWrappingInterface;
 
             m_shaderCode.clear();
             m_shaderCodeID.clear();
         }
         return *this;
     }
+    
+    void updateClassWrappingInterface()
+    {
+        if(m_language == GPU_LANGUAGE_MSL_2_0)
+        {
+            m_classWrappingInterface = std::unique_ptr<MetalShaderClassWrapper>(new MetalShaderClassWrapper);
+        }
+        else
+        {
+            m_classWrappingInterface = std::unique_ptr<NullGpuShaderClassWrapper>(new NullGpuShaderClassWrapper);
+        }
+    }
```

Comment:
The usual design is to create a factory method:
```
static std::unique_ptr<GpuShaderClassWrapper> CreateClassWrapper(m_language)
{
   switch language)
   {
       case MSL: return [...]
       [..]
   }
}
```
The use of `switch` helpers to detect missing enum case(s) at compile time (thanks to compiler's flag).